src/Form/Form.php

```diff
@@ -135,26 +135,57 @@ public function implementedEvents(): array
         return [];
     }
 
+    /**
+     * Set the schema for this form.
+     *
+     * @since 4.1.0
+     * @param \Cake\Form\Schema $schema The schema to set
+     * @return $this
+     */
+    public function setSchema(Schema $schema)
+    {
+        $this->_schema = $schema;
+
+        return $this;
+    }
+
+    /**
+     * Get the schema for this form.
+     *
+     * This method will call `_buildSchema()` when the schema
+     * is first built. This hook method lets you configure the
+     * schema or load a pre-defined one.
+     *
+     * @since 4.1.0
+     * @return \Cake\Form\Schema the schema instance.
+     */
+    public function getSchema(): Schema
+    {
+        if ($this->_schema === null) {
+            $this->_schema = $this->_buildSchema(new $this->_schemaClass());
+        }
+
+        return $this->_schema;
+    }
+
     /**
      * Get/Set the schema for this form.
      *
      * This method will call `_buildSchema()` when the schema
      * is first built. This hook method lets you configure the
      * schema or load a pre-defined one.
      *
+     * @deprecated 4.1.0 Use setSchema()/getSchema() instead.
      * @param \Cake\Form\Schema|null $schema The schema to set, or null.
      * @return \Cake\Form\Schema the schema instance.
      */
     public function schema(?Schema $schema = null): Schema
     {
-        if ($schema === null && empty($this->_schema)) {
-            $schema = $this->_buildSchema(new $this->_schemaClass());
-        }
-        if ($schema) {
-            $this->_schema = $schema;
+        if ($schema !== null) {
```

Comment:
Can we add a deprecation warning for this method?

Reply:
Throwing explicit deprecation warning in 4.1 would be quite annoying for users IMO.

Reply:
We should add those laters, like 4.3 or 4.4 maybe.

Reply:
Ok. Lets try not to get too many deprecations stacked up before we add warnings. It took me a good month to add all the warnings in preparation for 4.x.

Reply:
@markstory We should also consider simply not adding explicit deprecation warnings in future. A lot of devs use IDEs these days which already warn them when using deprecated methods plus static analysers like psalm can also do the same.

Reply:
I'm open to trying that. I've also started a rector ruleset for 4.1 in my fork. Once it has some more substance I'll open a pull request to upstream rector.